modules/dnn/include/opencv2/dnn/dnn.hpp

```diff
@@ -1251,6 +1251,56 @@ CV__DNN_INLINE_NS_BEGIN
     /** @overload */
     CV_EXPORTS_W void blobFromImagesWithParams(InputArrayOfArrays images, OutputArray blob, const Image2BlobParams& param = Image2BlobParams());
 
+    /** @brief Get rectangle coordinates in original image system from rectangle in blob coordinates.
+     *  @param rBlob rect in blob coordinates.
+     *  @param sizeOri original input image size inserted in blob.
+     *  @param param struct of Image2BlobParams, contains all parameters needed by processing of image to blob
+     *  @returns rectangle in original image coordinates.
+     */
+    CV_EXPORTS_W Rect blobRectToImageRect(Rect rBlob, Size sizeOri, Image2BlobParams param);
+
+    /** @brief Get rectangle coordinates in original image system from rectangle in blob coordinates.
+     *  @param rBlob rect in blob coordinates.
+     *  @param sizeOri original input image size inserted in blob.
+     *  @param param struct of Image2BlobParams, contains all parameters needed by processing of image to blob
+     *  @returns rectangle in original image coordinates.
+     */
+    CV_EXPORTS_W Rect2d blobRectToImageRect(Rect2d rBlob, Size sizeOri, Image2BlobParams param);
```

Comment:
Is a double precision a good choice for image coordinates? I understand that resize adds a scale factor but in terms of the final application for the result (drawing bounding box, crop sub-image) conversion to integers is an extra step for user.

Reply:
inference outputs in dnn are in float. I translated python code in c++ for [mediapipe models](https://github.com/opencv/opencv_zoo/tree/main/models/person_detection_mediapipe). All computation in python are in double. When you have to check results ( python versus c++) it becomes a problem.
pose estimation mediapipe use output of person detection mediapipe
May be @fengyuentau can give an advice

Reply:
I mean Rect2d returned from the function. Most likely used for drawing or ROI extraction. Both expect integer Rect, not double.